services/jitpack/jitpack-downloads.service.js

```diff
@@ -0,0 +1,81 @@
+'use strict'
+
+const Joi = require('joi')
+const { nonNegativeInteger } = require('../validators')
+const { BaseJsonService } = require('..')
+const { downloadCount } = require('../color-formatters')
+const { metric } = require('../text-formatters')
+
+const schema = Joi.object({
+  week: nonNegativeInteger,
+  month: nonNegativeInteger,
+}).required()
+
+const periodMap = {
+  dw: {
+    api_field: 'week',
+    suffix: '/week',
+  },
+  dm: {
+    api_field: 'month',
+    suffix: '/month',
+  },
+}
+
+module.exports = class JitpackDownloads extends BaseJsonService {
+  static get examples() {
+    return [
+      {
+        title: 'JitPack - Downloads',
+        namedParams: {
+          groupId: 'jitpack',
+          artifactId: 'maven-simple',
+          period: 'dm',
+        },
+        staticPreview: JitpackDownloads.render({
+          downloads: 14000,
+          period: 'dm',
+        }),
+        keywords: ['java', 'maven'],
+      },
+    ]
+  }
+
+  static get category() {
+    return 'downloads'
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'downloads' }
+  }
+
+  static get route() {
+    return {
+      base: 'jitpack/d',
+      pattern: ':groupId/:artifactId/:period(dw|dm)',
+    }
+  }
+
+  async handle({ groupId, artifactId, period }) {
+    const json = await this.fetch({ groupId, artifactId })
+    return this.constructor.render({
+      downloads: json[periodMap[period].api_field],
+      period,
+    })
+  }
+
+  async fetch({ groupId, artifactId }) {
+    return this._requestJson({
+      schema,
+      url: `https://jitpack.io/api/stats/com.github.${groupId}/${artifactId}`,
```

Comment:
I'm not familiar with jitpack myself, but is the `com.github.` prefix something we should always inject into the url, or was that supposed to be part of a groupId?

Reply:
Okay, okay, I think I've done it. It was my stupidity. I was just passing `period` to `render` method without its name. I'll push the new commits regarding to the changes that you've requested.

Reply:
I'm sorry. I meant it for the other review. Maybe I should sleep more. Whatever.

Back to the point, the former version implementation had the same structure. See [this](https://github.com/erayerdin/shields/blob/master/services/jitpack/jitpack.service.js#L47).

So I thought I should be consistent with it. Well, you are right at a point, what if a user deploys to Jitpack from another VCS host like Gitlab or Bitbucket. The problem is, if we change that behavior, people's Jitpack version badges might break. That wouldn't be good for them I think?

Reply:
>  On the other hand, if we change that behavior, people's Jitpack version badges might break. That wouldn't be good for them I think

While it is definitely true that we never want to break any existing users, updating the url here in the downloads service has no impact on any other services, including the version badge. 

Currently, the existing version badge presents the prompts to the user as `group id` and `artifact id`, but I think what it's really expecting is a GitHub user and repo. This existing behavior wasn't exactly clear to me from the UI/badge path. 

If I enter the group id myself (`com.github.jitpack`) then I will get an error, and similarly if I was using Jitpack with Gitlab/Bitbucket and tried to create a badge (`com.gitlab.foo`) I would get errors and in my opinion the _real_ reason behind the error would not be clear to me as a user:

![image](https://user-images.githubusercontent.com/13042488/54650184-4aed2880-4a7b-11e9-89ae-100b2cd1fdb6.png)

Given that Jitpack works with multiple VCS providers and that the Group Id does reflect the VCS (https://jitpack.io/docs/#other-git-hosts) I'd suggest three possible options:

1) Remove the manual `com.github` prefix we are adding to the new downloads badge. That way users could enter their respective group id's `com.github.jitpack`, `com.gitlab.foo`, `com.bitbucket.bar`, etc.
2) Update the route parameter names to more explicitly indicate that the user should enter a GitHub user name and repo (i.e. `:user/:repo`)
3) Add documentation to the examples on both the new Jitpack downloads badge and the existing Jitpack version badge to clearly convey that the badges will only work with GitHub 

I think I'd prefer option 1. Yes, going with option 1 would mean that in the short term users would have to enter the full group id for the downloads badge (`com.github.jitpack`) but just the GH user name for the version badge (`jitpack`). However, we could update the Jitpack version badge route in a follow up, including adding a redirect service to keep the existing badges working.

Curious what others think though

Reply:
I considered option 1 makes sense but, yet, I was again thinking about the existing users. So I came up with an idea, I don't know what you'd think about it.

Check this [regex](https://regex101.com/r/DXEuGq/1). At first, we will check if this expression maches the given value. If so, then we will render the badge with the given value.

If it does not match, though, we will prepend `com.github.` prefix to the value.

With this method, it will not break the existing users' badges and will work with both at the same time. If you want, I can implement this method to both `JitPackVersion` and `JitPackDownloads` classes. What do you think?

Reply:
Thanks @erayerdin 

Will review in more detail in a little while, but remember the JitPackDownloads badge is brand-new. There are no existing users of the downloads badge/it is not possible to break any existing users. 

The JitPack Version badge is untouched and not impacted by the downloads badge

Reply:
I have implemented it, it waits for your review.